Project-Resume-Graph/src/main/java/com/example/Project/Resume/Graph/service/ReadPdfService.java

```diff
@@ -1,327 +1,292 @@
-
                 if (location.isEmpty() && timePeriod.isEmpty()) {
                     Pattern pattern = Pattern.compile("(\\d*.years?.)?(\\d*.months?)");
                     Matcher matcher = pattern.matcher(description);
                     Boolean matchFound = matcher.find();
-
                     if (matchFound) {
                         timePeriod = matcher.group(0);
                         description = description.substring(timePeriod.length(), description.length() - 1);
                     }
                 }
-
                 if (!location.isEmpty() && timePeriod.isEmpty()) {
                     timePeriod = description;
                     description = "";
                 }
-
-                if (company != "" && company.charAt(company.length() - 1) == ' ')
-                    company = company.substring(0, company.length() - 1);
-                if (title != "" && title.charAt(title.length() - 1) == ' ')
-                    title = title.substring(0, title.length() - 1);
-                if (timePeriod != "" && timePeriod.charAt(timePeriod.length() - 1) == ' ')
-                    timePeriod = timePeriod.substring(0, timePeriod.length() - 1);
-                if (description != "" && description.charAt(description.length() - 1) == ' ')
-                    description = description.substring(0, description.length() - 1);
-
-                JSONObject jsonString = new JSONObject();
-                jsonString.put("company", company);
-                jsonString.put("title", title);
-                jsonString.put("time period", jsonStringService.getLongDuration(timePeriod));
-                jsonString.put("description", description);
-
-                positionsList.add(jsonString);
-
+                company = removeLastWhiteSpace(company);
+                title = removeLastWhiteSpace(title);
+                timePeriod = removeLastWhiteSpace(timePeriod);
+                description = removeLastWhiteSpace(description);
+                JSONObject jsonStringChild = new JSONObject();
+                jsonStringChild.put("company", company);
+                jsonStringChild.put("title", title);
+                jsonStringChild.put("time period", jsonStringService.getLongDuration(timePeriod));
+                jsonStringChild.put("description", description);
+                positionsList.add(jsonStringChild);
                 title = "";
                 description = "";
                 timePeriod = "";
                 location = "";
             }
-
-            if (styleIntigerValue1 == 12) {
-
+            if (styleFontSize == 12) {
                 company += element.text() + " ";
             }
-
-            if (styleIntigerValue1 == 11) {
-
+            if (styleFontSize == 11) {
                 title += element.text() + " ";
-
             }
-
-            if (styleIntigerValue1 == 10 && fontColor.matches("color:#b0b0b0")) {
+            if (styleFontSize == 10 && fontColor.matches("#b0b0b0")) {
```

Comment:
Please keep these hardcoded values in a property file or somewhere easy to change

Reply:
done